### PR TITLE
Use Numpy rng in FDEM tests

### PR DESCRIPTION
**Pull request by santisoler**
#### Summary

Replace the usage of the deprecated functions in `numpy.random` module for the Numpy's random number generator class and its methods, in most of the FDEM tests. Increase number of iterations for checking derivatives where needed.

#### PR Checklist
* [ ] If this is a work in progress PR, set as a Draft PR
* [ ] Linted my code according to the [style guides](https://docs.simpeg.xyz/content/getting_started/contributing/code-style.html).
* [ ] Added [tests](https://docs.simpeg.xyz/content/getting_started/practices.html#testing) to verify changes to the code.
* [ ] Added necessary documentation to any new functions/classes following the
      expect [style](https://docs.simpeg.xyz/content/getting_started/practices.html#documentation).
* [ ] Marked as ready for review (if this is was a draft PR), and converted
      to a Pull Request
* [ ] Tagged ``@simpeg/simpeg-developers`` when ready for review.

#### Reference issue

Part of the solution to #1289

### CODE REVIEW

**Comment by lheagy** [comment]
I can take a look into this fail

File: tests/em/fdem/muinverse/test_muinverse.py [hunk 1]
```diff
@@ -177,19 +178,21 @@ def fun(x):
                 lambda x: self.simulation.Jvec(mod, x),
             )
 
-        dx = np.random.rand(*mod.shape) * (mod.max() - mod.min()) * 0.01
+        rng = np.random.default_rng(seed=3321)
+        dx = rng.uniform(size=mod.shape) * (mod.max() - mod.min()) * 0.01
 
-        return tests.check_derivative(fun, mod, dx=dx, num=3, plotIt=False)
+        np.random.seed(1983)  # set a random seed for check_derivative
+        return tests.check_derivative(fun, mod, dx=dx, num=4, plotIt=False)
```
Comment (lheagy) [inline tests/em/fdem/muinverse/test_muinverse.py:185]: I increased the number of iterations here. Basically, the perturbations were a bit too big for us to see the second order behaviour, but by adding 1 (these ones are fast), you can see that the derivatives are indeed second order.

**Comment by lheagy** [comment]
@santisoler: this one should now be good to go!

File: tests/em/fdem/muinverse/test_muinverse.py (hunk 1 above)
Comment (santisoler) [reply tests/em/fdem/muinverse/test_muinverse.py:185]: Great! Thanks @lheagy!

**Pull request merged**